presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/RFC2822DateTimeFormatter.java

```diff
@@ -0,0 +1,111 @@
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import io.prestosql.spi.type.SqlDate;
+import io.prestosql.spi.type.SqlTime;
+import io.prestosql.spi.type.SqlTimeWithTimeZone;
+import io.prestosql.spi.type.SqlTimestamp;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+import static io.prestosql.plugin.base.type.DateTimes.NANOSECONDS_PER_MICROSECOND;
+import static io.prestosql.plugin.base.type.DateTimes.NANOSECONDS_PER_MILLISECOND;
+import static io.prestosql.plugin.base.type.DateTimes.getMicrosOfSecond;
+import static io.prestosql.plugin.base.type.DateTimes.getMillisOfSecond;
+import static io.prestosql.plugin.base.type.DateTimes.round;
+import static io.prestosql.plugin.base.type.DateTimes.scaleEpochMicrosToSeconds;
+import static io.prestosql.plugin.base.type.DateTimes.scaleEpochMillisToSeconds;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.NANOSECONDS_PRECISION;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.SECONDS_PRECISION;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.getNanosOfDay;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.scaleMillisToNanos;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.scalePicosToMillis;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.scalePicosToNanos;
+import static java.time.LocalDate.EPOCH;
+import static java.time.LocalTime.MIDNIGHT;
+import static java.time.ZoneOffset.UTC;
+
+public class RFC2822DateTimeFormatter
+        extends UnimplementedJsonDateTimeFormatter
+{
+    private static final DateTimeFormatter RFC_FORMATTER = DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss Z yyyy").withLocale(Locale.ENGLISH);
+
+    private static long roundNanosToSecondsPrecision(long nanos)
+    {
+        return round(nanos, NANOSECONDS_PRECISION - SECONDS_PRECISION);
+    }
+
+    private static LocalTime epochNanosToLocalTime(long epochNanos)
+    {
+        return LocalTime.ofNanoOfDay(getNanosOfDay(roundNanosToSecondsPrecision(epochNanos)));
+    }
+
+    @Override
+    public String formatDate(SqlDate value)
+    {
+        return RFC_FORMATTER.format(
+                ZonedDateTime.of(LocalDate.ofEpochDay(value.getDays()), MIDNIGHT, UTC));
+    }
+
+    @Override
+    public String formatTime(SqlTime value, int precision)
+    {
+        long epochNanos = roundNanosToSecondsPrecision(scalePicosToNanos(value.getPicos()));
+
+        return RFC_FORMATTER.format(
+                ZonedDateTime.of(EPOCH, epochNanosToLocalTime(epochNanos), UTC));
+    }
+
+    @Override
+    public String formatTimeWithZone(SqlTimeWithTimeZone value)
```

Comment:
I possibly can live with `formatDate` setting time part to zeros. But this one is imo a bit too far fetched. It feels to me that the file format which requires date part in it should not support `TIME`.

Reply:
I agree, supporting anything other than timestamps for RFC2822 seems ridiculous to me. The decoder supports them tho so I decided the encoder should too.

Reply:
let's not